Fix build 667.2

mindsbackyard opened the pull request:
Fixes the failing build 667.2

libsyntax changed:
- TyKind::Vec was renamed to TyKind::Slice
- TyKind::FixedLengthVec was renamed to TyKind::Array
- ast::LifetimeDef and ast::TyParam got a new 'attrs' field

Fixes tested with rust 1.14.0 nightly 2016-10-03
cargo build and cargo test run cleanly

mindsbackyard committed:
Fix renaming in libsyntax

syntax::ast::TyKind::Vec -> ...::Slice
syntax::ast::TyKind::FixedLengthVec -> ...::Array

mindsbackyard committed:
Fix renaming in libsyntax test cases for ty

mindsbackyard committed:
Fix new 'attrs' fields in ast::TyParam and ast::LifetimeDef

mindsbackyard committed:
Fix test cases with missing 'attrs' field

dtolnay commented:
This requires [a Syntex update](https://github.com/serde-rs/syntex/pull/95) before it will work on stable. Given that the Syntex work is going to take a while longer, we may want to release a version of Aster without Syntex support to at least unblock nightly users. @erickt?

mindsbackyard commented:
Yes, I just noticed the mistake. The syntex build fails but the libsyntax passes. Depending on how long it takes syntex to catch up maybe a conditional compilation would make it usable for both types of users (at the cost of maintaining that)?

mre commented:
You can monitor the syntex updates here: https://github.com/serde-rs/syntex/pull/95 (wip)

mindsbackyard committed:
Fix: restored syntex compatibility

Feature gated the parts where syntex currently diverges from libsyntax.
This is at best a temporary fix and probably has to be reverted once syntex is updated

mindsbackyard commented:
The latest commit feature-gates the problematic parts so all builds pass---it does the job but is not very nice.
It also duplicates the code (non-item attributes are still experimental). Probably that commit needs
to be reverted in the future.

dtolnay commented:
@erickt merged these commits (except the feature gate) in #115 and released 0.30.0.
